edge/pkg/devicetwin/dtmanager/communicate_test.go

```diff
@@ -0,0 +1,319 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dtmanager
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core/context"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dtcommon"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dtcontext"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dttype"
+)
+
+// TestStart is function to test Start().
+func TestStart(t *testing.T) {
+	mainContext := context.GetContext(context.MsgCtxTypeChannel)
+	dtContexts, _ := dtcontext.InitDTContext(mainContext)
+	heartChan := make(chan interface{}, 1)
+	receiveChan := make(chan interface{}, 1)
+	receiveChan <- &dttype.DTMessage{Action: "action", Identity: "identity", Msg: &model.Message{Content: "msg"}}
+	heartChan <- "stop"
+	tests := []struct {
+		name   string
+		Worker Worker
+		Group  string
+	}{
+		{
+			name: "StartTest",
+			Worker: Worker{
+				ReceiverChan:  receiveChan,
+				HeartBeatChan: heartChan,
+				DTContexts:    dtContexts,
+			},
+			Group: "",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cw := CommWorker{
+				Worker: test.Worker,
+				Group:  test.Group,
+			}
+			go cw.Start()
+		})
+	}
+}
+
+// TestDealSendToEdge is function to test dealsendToedge().
+func TestDealSendToEdge(t *testing.T) {
+	mainContext := context.GetContext(context.MsgCtxTypeChannel)
+	dtContexts, _ := dtcontext.InitDTContext(mainContext)
+	tests := []struct {
+		name     string
+		context  *dtcontext.DTContext
+		resource string
+		msg      interface{}
+		want     interface{}
+		wantErr  error
+	}{
+		{
+			name:     "dealSendToEdgeTest",
+			context:  dtContexts,
+			resource: "",
+			msg:      &model.Message{},
+			wantErr:  nil,
+			want:     nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, err := dealSendToEdge(test.context, test.resource, test.msg)
```

Comment:
the dealSendToEdge method always returns nil, nil. Are return values required here ? Please check and if not needed , adjust your code and test case accordingly

Reply:
Updated